### PR TITLE
Add Verus/Kani proofs and ExecPlan for LSH invariants (7.2.4/7.2.5)

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Adds a concrete Verus/Kani proof workflow for the clone-detector LSH invariants and accompanying tests; wires ExecPlan for 7.2.4/7.2.5 and updates documentation and tooling. This implements Stages B–I while preserving the runtime API.

## What this PR adds
- Verus proof: verus/clone_detector_lsh_config.rs (semantic model of LshConfig::new)
- Kani harness: crates/whitaker_clones_core/src/index/kani.rs (crate-private, bounded smoke harness)
- Proof tooling and wiring:
  - Makefile targets: verus-clone-detector, kani-clone-detector (in addition to existing verus and kani scaffolding)
  - Script scaffolding: scripts/install-kani.sh, scripts/run-kani.sh, scripts/run-verus.sh updated for grouped proof entry points
- ExecPlan documentation and wiring:
  - ExecPlan: docs/execplans/7-2-4-verus-proofs-for-lsh-config-new-invariants.md
- Documentation and test surface:
  - Documentation updates: docs/whitaker-clone-detector-design.md, docs/roadmap.md updates, and docs/developers-guide.md
  - Tests: extended unit tests and BDD coverage in min_hash_lsh feature area (zero bands, zero rows, invalid product, overflow edge)
  - Feature tests updated to cover new edge cases
- Artefacts this plan creates:
  - Verus proof: verus/clone_detector_lsh_config.rs
  - Kani harness: crates/whitaker_clones_core/src/index/kani.rs
  - Makefile and scripts: verus and Kani targets, install/run helpers
  - ExecPlan: docs/execplans/7-2-4-verus-proofs-for-lsh-config-new-invariants.md
  - Documentation updates: docs/whitaker-clone-detector-design.md, docs/roadmap.md, docs/developers-guide.md updates
  - Tests: extended unit tests and BDD coverage in min_hash_lsh feature area

## Scope and constraints
- Strict focus on roadmap items 7.2.4 and 7.2.5. Do not mark 7.2.6–7.2.8 as done in this change.
- Proof tooling remains sidecar-based (Verus for semantic invariants; Kani for bounded checks) per ADR 002.
- Preserve the existing runtime API shape; only crate-private proof seams may be introduced if needed.
- Build on the existing clone-detector crate (`crates/whitaker_clones_core`). Do not move `LshConfig` into a separate crate.
- If Kani Harnesses use `#[cfg(kani)]`, register that cfg in the crate's `unexpected_cfgs` allowlist.
- Keep each Rust source file focused and importable; avoid large monolithic files.
- Workspace-wide lint/version constraints maintained (e.g., `rstest-bdd` version).

## Implementation plan (highlights)
- Stage B: Add clone-detector proof sidecar tooling and Makefile targets
  - Extend scripts/run-verus.sh to group proofs by domain (clone-detector group)
  - Add install-kani.sh and run-kani.sh for Kani harnesses
  - Update Makefile with verus-clone-detector and kani-clone-detector targets
- Stage C: Extend unit tests (ZeroBands, ZeroRows, InvalidBandRowProduct, overflow edge)
- Stage D: Extend rstest-bdd coverage for min_hash_lsh config outcomes
- Stage E: Add Verus proof for LshConfig::new (verus/clone_detector_lsh_config.rs)
- Stage F: Add minimal Kani smoke harness (crates/whitaker_clones_core/src/index/kani.rs)
- Stage G: Update clone-detector design/roadmap docs
- Stage H: Gatekeeping and quality checks (fmt, lint, tests, docs)
- Stage I: Finalize ExecPlan after implementation

## Validation plan (evidence to collect)
- Run gates:
  - make fmt, make markdownlint, make nixie
  - make verus-clone-detector
  - make kani-clone-detector
  - make check-fmt, make lint, make test
- Optional broader proof smoke: make verus, make kani

## Acceptance criteria
- The ExecPlan document is added and clearly references 7.2.4 and 7.2.5.
- Stage B–I milestones are defined and traceable in the plan.
- Provisions for new proof tooling, targets, and tests are described (without changing runtime API yet).
- Design decisions are captured in the ExecPlan and related docs updated.

## Rationale
This change provides a concrete path to Verus-based proofs for LshConfig::new with a minimal Kani smoke harness, preserving API stability and keeping proof tooling separate from normal cargo/test flows as guided by ADR 002 and the roadmap.

## Related guidance
- ADR 002 on proof-workflow separation (Verus for semantic invariants; Kani for bounded checks).
- Existing LshConfig runtime contract constraints: bands > 0, rows > 0, bands * rows == MINHASH_SIZE (128).

## Progress
- Stage A–I completed for this turn; gates to be re-run in CI.

## Surprises & Discoveries
- The repository already includes a pinned Verus sidecar workflow; Kani harnesses are introduced here as a first-stage smoke gate.

## Task
- https://www.devboxer.com/task/8141455b-138d-4d5c-becb-8d6a210ecbad

### CODE REVIEW

File: scripts/run-kani.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+REPO_ROOT=$(CDPATH= cd -- "${SCRIPT_DIR}/.." && pwd)
+KANI_BIN=$("${SCRIPT_DIR}/install-kani.sh")
+MANIFEST_PATH="${REPO_ROOT}/crates/whitaker_clones_core/Cargo.toml"
+
+run_clone_detector_harnesses() {
+    "${KANI_BIN}" \
+        --manifest-path "${MANIFEST_PATH}" \
+        --default-unwind 4 \
+        --harness verify_lsh_config_new_smoke \
+        --harness verify_lsh_config_new_symbolic
```
[the remark this answers is not in the document]
Comment (leynos) [reply scripts/run-kani.sh]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

### ❌ Failed checks (1 warning)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Developer Documentation | ⚠️ Warning | New tooling and build requirements introduced in this PR are not documented in `docs/developers-guide.md`, the primary reference for developers. | Update `docs/developers-guide.md` to add a new section documenting the proof workflow tooling, including Makefile targets, scripts, Kani installation requirements, and examples. |

**Comment by leynos** [comment]
@coderabbitai are you satisfied that the introduced proof is substantive, rigorous, and well-founded?

File: verus/clone_detector_lsh_config.rs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+//! This module mirrors the runtime branch structure in `LshConfig::new` and
+//! `validate_product`: reject zero bands first, reject zero rows second, and
+//! then accept only non-zero inputs whose `checked_mul` product is exactly the
+//! fixed MinHash sketch width of 128.
+//!
+//! The sidecar proof still does not verify the compiled Rust body directly.
+//! In the current repository setup, linking this file to the production
+//! function would require trusted Verus assumptions for external code. The
+//! proof therefore stays honest about its scope: it proves a faithful model of
+//! the constructor logic, while Kani executes the real implementation.
```
[the remark this answers is not in the document]
Comment (leynos) [reply verus/clone_detector_lsh_config.rs:12]: @coderabbitai the engineer observes:

> Verified against the current tree. No fix was needed.

> [clone\_detector\_lsh\_config.rs](https://github.com/leynos/whitaker/blob/add-verus-proofs-lsh-new-invariants-5sefjo/verus/clone_detector_lsh_config.rs "Open verus/clone_detector_lsh_config.rs on GitHub (leynos/whitaker@add-verus-proofs-lsh-new-invariants-5sefjo)") is already demoted to a faithful model/contract record rather than pretending to verify the compiled constructor. The module-level comment states that explicitly, and the file proves `lsh_config_new_result` as a model of the runtime branch structure instead of claiming linkage to the real Rust body. The supporting docs say the same thing in [developers-guide.md](https://github.com/leynos/whitaker/blob/add-verus-proofs-lsh-new-invariants-5sefjo/docs/developers-guide.md "Open docs/developers-guide.md on GitHub (leynos/whitaker@add-verus-proofs-lsh-new-invariants-5sefjo)"), [whitaker-clone-detector-design.md](https://github.com/leynos/whitaker/blob/add-verus-proofs-lsh-new-invariants-5sefjo/docs/whitaker-clone-detector-design.md "Open docs/whitaker-clone-detector-design.md on GitHub (leynos/whitaker@add-verus-proofs-lsh-new-invariants-5sefjo)"), and [7-2-4-verus-proofs-for-lsh-config-new-invariants.md](https://github.com/leynos/whitaker/blob/add-verus-proofs-lsh-new-invariants-5sefjo/docs/execplans/7-2-4-verus-proofs-for-lsh-config-new-invariants.md "Open docs/execplans/7-2-4-verus-proofs-for-lsh-config-new-invariants.md on GitHub (leynos/whitaker@add-verus-proofs-lsh-new-invariants-5sefjo)").

> So the review comment is stale relative to the current code: the sidecar no longer presents itself as a direct proof of `LshConfig::new`, and it already documents why `external_fn_specification` / `external_body` are not being used here.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for the following concern and provide an AI coding agent prompt for the fix:

### ❌ Failed checks (1 warning)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing | ⚠️ Warning | Core invariants tested via unit and BDD tests in CI, but shell scripts and proof tooling infrastructure lack automated test coverage. | Implement BATS tests for shell scripts and integrate Kani/Verus proof execution into CI as required or documented optional verification steps. |
